File: box/src/test/java/de/qabel/box/storage/BoxVolumeLocalTest.java
package de.qabel.box.storage;

import org.apache.commons.io.FileUtils;
import org.junit.Test;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.IsEqual.equalTo;

public class BoxVolumeLocalTest extends BoxVolumeTest {

    private Path tempFolder;
    private LocalReadBackend readBackend;

    @Override
    protected StorageReadBackend getReadBackend() {
        return readBackend;
    }

    @Override
    public void setUpVolume() throws IOException {
        tempFolder = Files.createTempDirectory("");
        prefix = "";

        readBackend = new LocalReadBackend(tempFolder);
        volume = new BoxVolumeImpl(readBackend,
                new LocalWriteBackend(tempFolder),
                keyPair, deviceID, new File(System.getProperty("java.io.tmpdir")), "");
        volume2 = new BoxVolumeImpl(new LocalReadBackend(tempFolder),
                new LocalWriteBackend(tempFolder),
                keyPair, deviceID2, new File(System.getProperty("java.io.tmpdir")), "");
    }

    @Override
    protected void cleanVolume() throws IOException {
        FileUtils.deleteDirectory(tempFolder.toFile());
    }

    @Test
    public void rootRefNotChanged() throws Exception {
        assertThat(volume.getRootRef(), equalTo("300c9c96-03b9-2a4b-39ed-3958bf924011"));
    }
}

File: box/src/test/java/de/qabel/box/storage/BoxVolumeLocalTest.kt
package de.qabel.box.storage

import org.apache.commons.io.FileUtils
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.core.IsEqual.equalTo
import org.junit.Test
import java.io.File
import java.io.IOException

class BoxVolumeLocalTest : BoxVolumeTest() {
    private val tempFolder: File by lazy { createTempDir("longerPrefix") }

    override val readBackend: StorageReadBackend by lazy { LocalReadBackend(tempFolder) }

    @Throws(IOException::class)
    public override fun setUpVolume() {
        prefix = ""

        volume = BoxVolumeImpl(readBackend,
                LocalWriteBackend(tempFolder),
                keyPair, deviceID, File(System.getProperty("java.io.tmpdir")), "")
        volume2 = BoxVolumeImpl(LocalReadBackend(tempFolder),
                LocalWriteBackend(tempFolder),
                keyPair, deviceID2, File(System.getProperty("java.io.tmpdir")), "")
    }

    @Throws(IOException::class)
    override fun cleanVolume() {
        FileUtils.deleteDirectory(tempFolder)
    }

    @Test
    @Throws(Exception::class)
    fun rootRefNotChanged() {
        assertThat(volume.rootRef, equalTo("300c9c96-03b9-2a4b-39ed-3958bf924011"))
    }
}
